src/analysis/normed_space/operator_norm.lean

```diff
@@ -381,14 +382,89 @@ lemma op_norm_smul_le {𝕜' : Type*} [normed_field 𝕜'] [normed_space 𝕜' F
   end))
 
 /-- Continuous linear maps themselves form a seminormed space with respect to
-    the operator norm. -/
-instance to_seminormed_add_comm_group : seminormed_add_comm_group (E →SL[σ₁₂] F) :=
+the operator norm. This is only a temporary definition because we want to replace the topology
+with `continuous_linear_map.topological_space` to avoid diamond issues.
+See Note [forgetful inheritance] -/
+private def tmp_seminormed_add_comm_group : seminormed_add_comm_group (E →SL[σ₁₂] F) :=
 add_group_seminorm.to_seminormed_add_comm_group
 { to_fun := norm,
   map_zero' := op_norm_zero,
   add_le' := op_norm_add_le,
   neg' := op_norm_neg }
 
+private def tmp_pseudo_metric_space : pseudo_metric_space (E →SL[σ₁₂] F) :=
+tmp_seminormed_add_comm_group.to_pseudo_metric_space
+
+private def tmp_uniform_space : uniform_space (E →SL[σ₁₂] F) :=
+tmp_pseudo_metric_space.to_uniform_space
+
+private def tmp_topological_space : topological_space (E →SL[σ₁₂] F) :=
+tmp_uniform_space.to_topological_space
+
+section tmp
+
+local attribute [-instance] continuous_linear_map.topological_space
+local attribute [-instance] continuous_linear_map.uniform_space
+local attribute [instance] tmp_seminormed_add_comm_group
+
+private lemma tmp_topological_add_group : topological_add_group (E →SL[σ₁₂] F) :=
+infer_instance
+
+private lemma tmp_closed_ball_div_subset {a b : ℝ} (ha : 0 < a) (hb : 0 < b) :
+  closed_ball (0 : E →SL[σ₁₂] F) (a / b) ⊆
+  {f | ∀ x ∈ closed_ball (0 : E) b, f x ∈ closed_ball (0 : F) a} :=
+begin
+  intros f hf x hx,
+  rw mem_closed_ball_zero_iff at ⊢ hf hx,
+  calc ∥f x∥
+      ≤ ∥f∥ * ∥x∥ : le_op_norm _ _
+  ... ≤ (a/b) * b : mul_le_mul hf hx (norm_nonneg _) (div_pos ha hb).le
+  ... = a : div_mul_cancel a hb.ne.symm
+end
+
+end tmp
+
+private theorem tmp_topology_eq :
+  (tmp_topological_space : topological_space (E →SL[σ₁₂] F)) =
+  infer_instance :=
```

Comment:
Could you instead give explicitly the two topologies? It would be much more readable.